services/navigation/builtin/builtin.go

```diff
@@ -399,14 +400,17 @@ func (svc *builtIn) startWaypoint(ctx context.Context, extra map[string]interfac
 			return svc.waypointReached(ctx)
 		}
 
-		// loop until no waypoints remaining
+		// do not exit loop - even if there are no waypoints
 		for {
+		WP:
 			if ctx.Err() != nil {
 				return
 			}
 
 			wp, err := svc.store.NextWaypoint(ctx)
-			if err != nil {
+			if err != nil && err.Error() == "no more waypoints" {
+				goto WP
+			} else if err != nil {
```

Comment:
Not sure if we should even be exiting here.  If we have an error from this function but we can't return it, it will quietly shut down the navigation service loop.  I'm thinking instead we should be just ignoring the error and continuing.  This would mean we wouldn't need to check the type of the error